docs/IR.md

```diff
@@ -271,27 +271,54 @@ Name|Type|Description|Spec Variant|
 |records|Record[]|A list of record values|ONNX-ML|
 |nullables|Nullable[]|A list of nullable values|ONNX-ML|
 |sparses|SparseTensor[]|A list of sparse tensors|ONNX-ML|
-|types|TypeProto[]|A list of types|ONNX-ML|
 
-The properties ‘name’ and ‘type’ are required on all attributes, and ‘doc_string’ SHOULD be used on all attributes. An attribute MUST have only one of the value-carrying properties.
+The properties ‘name’ and ‘type’ are required on all attributes, and ‘doc_string’ __SHOULD__ be used on all attributes. An attribute __MUST__ have only one of the value-carrying properties.
 
 __Note__: ONNX-ML will likely introduce the ability to define custom operators that are higher-order, i.e. accept one or more graphs as  attributes. In order to statically represent the expected signature of such higher-order custom operators, we will likely have to introduce the notion of a function signature into the attribute type system.
 
+#### Type Attributes
+
+For some operators, it is necessary to define the respective types of some of their outputs using attributes. 
+
+Such attributes __MUST__ be encoded as strings using the following grammar:
+
+```
+<type> ::= <data_type> |
+           tensor(<data_type>) |
+           sparse(<data_type>) |
+           seq(<type>) |
+           map(<data_type> , <type>) |
```

Comment:
Should we define <key_type> to be different from <data_type> ? E.g., having floats/doubles as a keytype is a bad idea usually, isn't it?

Reply:
This grammar just follows what the TypeProto definition for maps says.